src/transformers/models/t5/modeling_t5.py

```diff
@@ -642,10 +670,12 @@ def forward(
         hidden_states, present_key_value_state = self_attention_outputs[:2]
         attention_outputs = self_attention_outputs[2:]  # Keep self-attention outputs and relative position weights
 
+        detect_overflow(hidden_states, "T5Block after T5LayerSelfAttention")
+
         # clamp inf values to enable fp16 training
-        if hidden_states.dtype == torch.float16 and torch.isinf(hidden_states).any():
-            clamp_value = torch.finfo(hidden_states.dtype).max - 1000
-            hidden_states = torch.clamp(hidden_states, min=-clamp_value, max=clamp_value)
+        # if hidden_states.dtype == torch.float16 and torch.isinf(hidden_states).any():
+        #    clamp_value = torch.finfo(hidden_states.dtype).max - 1000
+        #    hidden_states = torch.clamp(hidden_states, min=-clamp_value, max=clamp_value)
```

Comment:
so please let me know if this is still needed. 

Without having a test that this feature was added with it's hard to know whether this will be removing some protection. Let's try to write tests when adding new features if it's doable and not too difficult.